### PR TITLE
Kustomize

### PR DESCRIPTION
**Pull request by anthr76**
This further inlines Flux's idea of GitOps with the Flux v2 gotk. With these changes multi-cluster multi-tenancy becomes much simpler. 

More information can be found on @xUnholy repo.

https://github.com/xUnholy/k8s-gitops/pull/270

gitops

What's out of scope on this PR:

- New secret mechanism
- Removing NWK1 overlays from base
- Automation cleanup with renovate etc
- Extensive validation

The ladder will be taken care of in future PRs

### CODE REVIEW

File: gitops/core/base/crds/traefik/crds.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  interval: 5m
+  prune: false
+  sourceRef:
+    kind: GitRepository
+    name: traefik-crd-source
+  healthChecks:
+    - apiVersion: helm.toolkit.fluxcd.io/v1beta1
+      kind: HelmRelease
+      name: traefik
+      namespace: traefik
```
Comment (xUnholy) [inline gitops/core/base/crds/traefik/crds.yaml:33]: AFAIK healthchecks are supports to be for the resource the kustomize deploys and manages, not necessarily for a resource created in another kustomize, similar to how healthchecks work in deployments/pods

File: gitops/namespaces/base/networking/external-dns/kustomization.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+- external
+- internal
```
Comment (xUnholy) [inline gitops/namespaces/base/networking/external-dns/kustomization.yaml:5]: Are these directories?

Comment (xUnholy) [reply gitops/namespaces/base/networking/external-dns/kustomization.yaml:5]: It's hard to tell in the PR as I'm only having a quick glance over

File: gitops/namespaces/overlays/kind/patch.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+      version: 9.14.2
+      sourceRef:
+        kind: HelmRepository
+        name: traefik-charts
+        namespace: flux-system
+      interval: 5m
+  values:
+    deployment:
+      kind: Deployment
+    additionalArguments:
```
Comment (xUnholy) [inline gitops/namespaces/overlays/kind/patch.yaml:22]: This looks like an empty value for the `additionalArguments` field

File: gitops/namespaces/base/networking/external-dns/kustomization.yaml (hunk 2 above)
Comment (anthr76) [reply gitops/namespaces/base/networking/external-dns/kustomization.yaml:5]: These are indeed directories and would like to know if it's necessary. In my short testing I have one externaldns deployed like:

```
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization
resources:
- ../../base/networking
- ../../base/traefik/
- ../../base/traefik/traefik
- ../../base/flux-system
- ../../base/networking/external-dns/internal
```

File: gitops/namespaces/overlays/kind/patch.yaml (hunk 3 above)
Comment (anthr76) [reply gitops/namespaces/overlays/kind/patch.yaml:22]: Yep left this false to override my `additionalArguments` in base.

File: gitops/core/base/crds/traefik/crds.yaml (hunk 1 above)
Comment (anthr76) [reply gitops/core/base/crds/traefik/crds.yaml:33]: Thanks for this. Before merging i'm going to read up on the healthchecks but shouldn't be needed

**Pull request merged**